src/Invio.Immutable/DoublePropertyHandlerProvider.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Linq;
+using System.Reflection;
+
+namespace Invio.Immutable {
+    /// <summary>
+    ///   This <see cref="IPropertyHandlerProvider" /> will create an appropriately
+    ///   configured <see cref="DoublePropertyHandler" /> if the <see cref="PropertyInfo" />
+    ///   provided has a <see cref="PropertyInfo.PropertyType" /> of type
+    ///   <see cref="Double" />.
+    /// </summary>
+    public class DoublePropertyHandlerProvider : PropertyHandlerProviderBase {
+        /// <summary>
+        ///   Checks to see if the <see cref="PropertyInfo" /> has a
+        ///   <see cref="PropertyInfo.PropertyType" /> of type <see cref="Single" /> to determine
+        ///   if the <see cref="PropertyInfo" /> can be use to create a
+        ///   <see cref="SinglePropertyHandler" />.
+        /// </summary>
+        /// <inheritdoc />
+        protected override bool IsSupportedImpl(PropertyInfo property) {
+            return base.IsSupportedImpl(property) && property.PropertyType == typeof(Double);
+        }
+
+        /// <summary>
+        ///   Creates an appropriately configured instance of <see cref="DoublePropertyHandler" />
+        ///   whenever the <see cref="PropertyInfo.PropertyType" /> on the
+        ///   <paramref name="property" /> provided is of type <see cref="Double" />.
+        /// </summary>
+        /// <param name="property">
+        ///   A <see cref="PropertyInfo" /> that references a property that exists on a class that
+        ///   derives from <see cref="ImmutableBase{TImmutable}" /> that has a
+        ///   <see cref="PropertyInfo.PropertyType" /> of type <see cref="Double" />.
+        /// </param>
+        /// <returns>
+        ///   A <see cref="DoublePropertyHandler" /> that will calculate hash codes and determine
+        ///   equality for Double stored <paramref name="property" /> within instances of
+        ///   <see cref="ImmutableBase{TImmutable}" />.
+        /// </returns>
+        protected override IPropertyHandler CreateImpl(PropertyInfo property) {
+
+            if (TryGetAttribute(property, out var attribute)) {
+                return new DoublePropertyHandler(
+                    property,
+                    attribute.Precision,
+                    attribute.SignificantFigures
+                );
+            } else if (TryGetAttribute(property.ReflectedType.GetTypeInfo(), out attribute)) {
+                return new DoublePropertyHandler(
+                    property,
+                    attribute.Precision,
+                    attribute.SignificantFigures
+                );
+            }
+
+            return new DefaultPropertyHandler(property);
```

Comment:
This line of code will need to be kept synchronized with the `DefaultPropertyHandlerProvider`, as you've decided to return the default implementation when the double property is not decorated. Instead, I would suggest one of two options:
1. Return a `DoublePropertyHandler` with a specific set of default values that makes sense based upon your testing and on-the-ground initial experience with properties of type double.
2. Return a `false` from `IsSupportedImpl(PropertyInfo)` when the property is not decorated.

The same comment would apply for the `SinglePropertyHandlerProvider`.